vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -249,13 +250,22 @@ func (d *VolumeDriver) Create(r volume.Request) volume.Response {
 
 	dev, errAttach := d.ops.Attach(r.Name, nil)
 	if errAttach != nil {
-		log.WithFields(log.Fields{"name": r.Name,
-			"error": errAttach}).Error("Attach volume failed, removing the volume ")
-		errRemove := d.ops.Remove(r.Name, nil)
-		if errRemove != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errRemove}).Warning("Remove volume failed ")
+		if strings.Contains(errAttach.Error(), internalError) {
```

Comment:
I've 2 concerns here;
-special casing error strings
-band-aid fix to retry attach

Why not remove special else (ret != 0 && err == NULL) and  re-try [here](https://github.com/vmware/docker-volume-vsphere/blob/master/vmdk_plugin/drivers/vmdk/vmdkops/esx_vmdkcmd.go#L125)